### PR TITLE
Upgrade to SQLA v2

### PR DESCRIPTION
**Pull request by lmmx**
## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes

### CODE REVIEW

**Comment by lmmx** [comment]
Error arising in the graph demo

```py
        try:
>           await connected
E           asyncpg.exceptions.InvalidPasswordError: password authentication failed for user "test_postgres"
```

The username/password are the same, and read from the config file: so if the username is read then the password should be too (unknown what is going wrong here)

**Comment by Dreamsorcerer** [comment]
> Instance <User at 0x7fc2ce19ff40> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: https://sqlalche.me/e/20/bhk3)

Looks like you're onto something!

**Comment by Dreamsorcerer** [comment]
Woo, the graphql demo is finally passing again. Should be easy to mop up the remaining demos now. I'll help out this week if it's not obvious what's wrong with the others.

**Comment by lmmx** [comment]
I think I caught the last of them! All tests pass locally now, waiting to see the CI result :eyes:

**Comment by lmmx** [comment]
Hm, OK that last commit fixed the linting but somehow broke the tests (which were passing on CI)

**Comment by Dreamsorcerer** [comment]
Seems to be passing?

Probably will need some cleanup regardless though, I'll try and look over the code changes tomorrow.

**Comment by lmmx** [comment]
Ah, I'd accidentally `git stash`ed my `.venv` and was wondering why the tests no longer passed locally :sweat_smile: 

Yep this looks done now, marking ready for review!

I see one error when upgrading to 3.11 (to do with `asyncio.coroutine`) but that can be for another issue

**Comment by Dreamsorcerer** [comment]
It bugs me that the error in the tests is not failing the CI. Not sure why...

File: demos/polls/tests/test_integration.py [hunk 1]
```diff
@@ -43,12 +43,13 @@ async def test_vote(cli, tables_and_data):
         )
         assert response.status == 200
 
-        res = await conn.execute(
-            choice.select(
-            ).where(choice.c.question_id == question_id)
-             .where(choice.c.choice_text == choice_text)
+    async with cli.server.app['db'].begin() as sess:
```
Comment (Dreamsorcerer) [inline demos/polls/tests/test_integration.py:46]:
```suggestion
    async with cli.server.app["db"].begin() as sess:
```

File: docs/tutorial.rst [hunk 2]
```diff
@@ -203,7 +213,6 @@ clean and short::
 Next, load the config into the application:
 
 .. code-block:: python
-    :emphasize-lines: 9
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:216]:
```suggestion
:emphasize-lines: 8

```

File: docs/tutorial.rst [hunk 3]
```diff
@@ -203,7 +213,6 @@ clean and short::
...
 
     # aiohttpdemo_polls/main.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:217]:
```suggestion
```

File: docs/tutorial.rst [hunk 4]
```diff
@@ -254,105 +263,66 @@ We will use SQLAlchemy to describe database schema for two related models,
...
-
-
-    But it doesn't give much benefits later on. SQLAlchemy ORM doesn't work in
-    asynchronous style and as a result ``aiopg.sa`` doesn't support related ORM
-    expressions such as ``Question.query.filter_by(question_text='Why').first()``
-    or ``session.query(TableName).all()``.
+.. code-block:: python
 
-    You still can make ``select`` queries after some code modifications:
+    # aiohttpdemo_polls/db.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:270]:
```suggestion
```

File: docs/tutorial.rst [hunk 5]
```diff
@@ -254,105 +263,66 @@ We will use SQLAlchemy to describe database schema for two related models,
...
+        )
 
 
 Now we need to create tables in database as it was described with sqlalchemy.
-Helper script can do that for you. Create a new file ``init_db.py`` in project's root::
+Helper script can do that for you. Create a new file ``init_db.py`` in project's root:
+
+.. code-block:: python
 
     # polls/init_db.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:301]:
```suggestion
```

File: docs/tutorial.rst [hunk 6]
```diff
@@ -363,10 +333,11 @@ Helper script can do that for you. Create a new file ``init_db.py`` in project's
     A more advanced version of this script is mentioned in :ref:`aiohttp-demos-polls-preparations-database` notes.
 
 
-Install the ``aiopg[sa]`` package (it will pull ``sqlalchemy`` alongside) to interact with the database,
+Install both the ``asyncpg`` (we'll be using it later) and ``sqlalchemy`` packages to interact with the database,
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:336]:
```suggestion
Install both the ``asyncpg`` and ``sqlalchemy`` packages to interact with the database,
```

File: docs/tutorial.rst [hunk 7]
```diff
@@ -407,10 +378,19 @@ Creating connection engine
...
+
+    DSN = "postgresql+asyncgp://{user}:{password}@{host}:{port}/{database}"
+    
+    async def pg_context(app):
+        engine = await create_async_engine(DSN.format(**app['config']['postgres']))
+        app["db"] = async_sessionmaker(engine)
+
+        yield
+
+        await engine.dispose()
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:393]:
```suggestion
.. literalinclude:: ../demos/polls/aiohttpdemo_polls/db.py
  :pyobject: pg_context
```

File: docs/tutorial.rst [hunk 8]
```diff
@@ -436,61 +416,44 @@ Complete files with changes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 .. code-block:: python
-    :emphasize-lines: 33, 47
-
+    
     # aiohttpdemo_polls/db.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:420]:
```suggestion
```

File: docs/tutorial.rst [hunk 9]
```diff
@@ -436,61 +416,44 @@ Complete files with changes
...
-        app['db'].close()
-        await app['db'].wait_closed()
+            yield
 
+            await engine.dispose()
 
 .. code-block:: python
-    :emphasize-lines: 6, 11
 
     # aiohttpdemo_polls/main.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:458]:
```suggestion
```

File: docs/tutorial.rst [hunk 10]
```diff
@@ -542,28 +505,54 @@ library first:
 After installing, setup the library:
 
 .. code-block:: python
-    :emphasize-lines: 3, 4, 12, 13
 
     # aiohttpdemo_polls/main.py
     from aiohttp import web
-    import aiohttp_jinja2
-    import jinja2
+    import aiohttp_jinja2, jinja2
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:511]:
```suggestion
    import aiohttp_jinja2
    import jinja2
```

File: docs/tutorial.rst [hunk 11]
```diff
@@ -542,28 +505,54 @@ library first:
...
     from db import pg_context
 
     app = web.Application()
-    app['config'] = config
+    app["config"] = config
     aiohttp_jinja2.setup(app,
-        loader=jinja2.FileSystemLoader(str(BASE_DIR / 'aiohttpdemo_polls' / 'templates')))
+        loader=jinja2.FileSystemLoader(
+            str(BASE_DIR / 'aiohttpdemo_polls' / 'templates')
+            ))
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:522]:
```suggestion
        loader=jinja2.FileSystemLoader(str(BASE_DIR / "aiohttpdemo_polls" / "templates")))
```

File: docs/tutorial.rst [hunk 12]
```diff
@@ -542,28 +505,54 @@ library first:
 After installing, setup the library:
 
 .. code-block:: python
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:507]:
```suggestion
.. code-block:: python
    :emphasize-lines: 3, 4, 12, 13
```

File: docs/tutorial.rst [hunk 13]
```diff
@@ -587,28 +576,47 @@ dict with page content, the ``aiohttp_jinja2.template`` decorator
 processes the dict using the jinja2 template renderer.
 
 .. code-block:: python
-    :emphasize-lines: 5, 11
 
     # aiohttpdemo_polls/views.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:580]:
```suggestion
```

File: docs/tutorial.rst [hunk 14]
```diff
@@ -587,28 +576,47 @@ dict with page content, the ``aiohttp_jinja2.template`` decorator
 processes the dict using the jinja2 template renderer.
 
 .. code-block:: python
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:578]:
```suggestion
.. code-block:: python
    :emphasize-lines: 6, 10
```

File: docs/tutorial.rst [hunk 15]
```diff
@@ -587,28 +576,47 @@ dict with page content, the ``aiohttp_jinja2.template`` decorator
...
 
 Run the server and you should see a question decorated in html list element.
 
 
-Let's add more views:
+Now let's add more views:
+
+Update ``views.py`` and ``db.py`` one last time, to:
 
 .. literalinclude:: ../demos/polls/aiohttpdemo_polls/views.py
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:600]:
```suggestion
.. literalinclude:: ../demos/polls/aiohttpdemo_polls/views.py
  :pyobject: poll
```

File: docs/tutorial.rst [hunk 16]
```diff
@@ -587,28 +576,47 @@ dict with page content, the ``aiohttp_jinja2.template`` decorator
...
-  :pyobject: poll
+
+.. literalinclude:: ../demos/polls/aiohttpdemo_polls/db.py
+
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:603]:
```suggestion
```

File: docs/tutorial.rst [hunk 17]
```diff
@@ -587,28 +576,47 @@ dict with page content, the ``aiohttp_jinja2.template`` decorator
...
+
+Add the following to ``routes.py``:
+
+.. code-block:: python
+
+    app.router.add_get('/poll/{question_id}', poll, name='poll')
+    app.router.add_get('/poll/{question_id}/results',
+                       results, name='results')
+    app.router.add_post('/poll/{question_id}/vote', vote, name='vote')
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:612]:
```suggestion
    app.router.add_get("/poll/{question_id}", poll, name="poll")
    app.router.add_get("/poll/{question_id}/results",
                       results, name="results")
    app.router.add_post("/poll/{question_id}/vote", vote, name="vote")
```

File: docs/tutorial.rst [hunk 18]
```diff
@@ -624,13 +632,54 @@ proxy like NGINX or using CDN services.
...
+
+.. code-block:: python
+
+    import pathlib
+    PROJECT_ROOT = pathlib.Path(__file__).parent
+    
+    def setup_static_routes(app):
+        app.router.add_static('/static/',
+                              path=PROJECT_ROOT / 'static',
+                              name='static')
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:647]:
```suggestion
.. literalinclude:: ../demos/polls/aiohttpdemo_polls/routes.py
  :pyobject: setup_static_routes
```

File: docs/tutorial.rst [hunk 19]
```diff
@@ -203,9 +213,8 @@ clean and short::
 Next, load the config into the application:
 
 .. code-block:: python
-    :emphasize-lines: 9
+:emphasize-lines: 8
```
Comment (Dreamsorcerer) [inline docs/tutorial.rst:216]:
```suggestion
    :emphasize-lines: 8
```

**Comment by Dreamsorcerer** [comment]
Thanks a lot for that! If you feel like doing some more work on the demos, there are a few more issues open which should be smaller tasks than this one.

**Comment by Dreamsorcerer** [comment]
#198 would be a good one to get resolved.

**Pull request merged**